elastic/kibana
# [Fleet] Couple agent and package policies spaces

## Summary

Resolve #194099 

That PR couple agent and package policies and do not allow to add reusable integration policies, to avoid bug where we expect agent policies and integrations policies spaces to be the same.

All the validations are happening server side only

## UI Changes 

<img width="1785" alt="Screenshot 2024-10-23 at 9 59 15 AM" src="https://github.com/user-attachments/assets/1985acf2-a147-4ef5-9ed7-6b11abd13ed3">


## How to test?

With space awareness enabled (feature flag + API call)

* Create an agent policy in multiple space, verify you cannot add a reusable integration policy to that agent policy
* Create a new integration policy, verify the policy is accessible in multiple space


## Todo

- [x] Test agent policy validation 
- [x] Test package policy validation
